boards/cats_rev1Pro/src/cli/cli_commands.c

```diff
@@ -533,24 +533,61 @@ static void cli_cmd_dump_flight(const char *cmd_name, char *args) {
   }
 }
 
+/* flight_parse <flight_idx> [--filter <RECORDER TYPE>...] */
 static void cli_cmd_parse_flight(const char *cmd_name, char *args) {
-  /* TODO - count how many files in a directory here */
-  char *endptr;
-  uint32_t flight_idx = strtoul(args, &endptr, 10);
+  char *ptr = strtok(args, " ");
 
-  if (args != endptr) {
-    // A number was found
-    if (flight_idx > flight_counter) {
-      cli_print_linef("\nFlight %lu doesn't exist", flight_idx);
-      cli_print_linef("Number of recorded flights: %lu", flight_counter);
+  uint32_t flight_idx = 0;
+  rec_entry_type_e filter_mask = 0;
+
+  /* Read flight_idx */
+  if (ptr != NULL) {
+    char *endptr;
+    flight_idx = strtoul(ptr, &endptr, 10);
+
+    if (args != endptr) {
+      // A number was found
+      if (flight_idx > flight_counter) {
+        cli_print_linef("\nFlight %lu doesn't exist", flight_idx);
+        cli_print_linef("Number of recorded flights: %lu", flight_counter);
+      }
     } else {
-      cli_print_linefeed();
-      parse_recording(flight_idx);
+      cli_print_linef("\nBad flight index argument: %s!", ptr);
+      return;
     }
   } else {
-    cli_print_line("\nArgument not provided!");
+    cli_print_line("\nFlight index not provided!");
+    return;
+  }
+
+  /* Read filter command */
+  ptr = strtok(NULL, " ");
```

Comment:
This line I dont understand .... you are breaking up NULL in different tokens?

Reply:
https://www.tutorialspoint.com/c_standard_library/c_function_strtok.htm

check this example out ^

strtok is weird, it keeps its internal state and you have to pass it NULL to continue processing the args string.